### DISCUSSION TITLE
No longer able to connect to CS50 codespaces

### DISCUSSION THREAD

**Discussion by ate26x** [community/community#65413] [Codespaces] [answered]
### No longer able to connect to CS50 Codespaces

Bug 

### Body

I created codespace via CS50.dev months ago, and have been using it since. However, suddenly I am unable to do so since two weeks ago. After logging on github (via cs50.dev), it shows "Codespace creation failed", the status "connecting...", and stalls. 

Screenshot here: https://i.stack.imgur.com/FTD1y.png

Alternatively, per this guide: https://cs50.readthedocs.io/code/ , I installed VS Code desktop, github codespace extensions, and tried signing in. This failed, with the message "Failed to connect to the codespace through the relay: invalid relay SAS", and "Source: Github Codespaces (Extension).

Screenshot on VS Code desktop: 
https://i.stack.imgur.com/o6L8g.png

_* If you have reservations clicking these links, please see the same on stack overflow (below)_

Tried asking around Gitter, however, did not help understanding, and hence resolving this.

I have created these tickets on the same via:
**i) VS Code support on github:** https://github.com/microsoft/vscode/issues/191574
**ii) github codespaces support:** https://support.github.com/ticket/personal/0/2310738
**iii) stackoverflow:** https://stackoverflow.com/questions/76998936/no-longer-able-to-connect-to-codespace

Also, I referred these posts on this forum:
**i) community discussion post #62627** (https://github.com/orgs/community/discussions/62627)
Tried restarting as it suggests, but no joy

**ii) community discussion post #15488** (https://github.com/orgs/community/discussions/15488)
This seems similar, however, for me, codespaces on the web shows "Codespace Creation failed" instead of "Setting up your Codespace "

Can anyone explain what's going on, and what I may do to resolve this? Thanks!

**Comment by ate26x** [comment, marked as answer]
Github support team helped and solved the issue. Here's how:

i. Navigate to https://github.com/code50/{CS50 codespace id} 
ii. Click on Code > Codespaces > + 
iii. Wait for codespace to open

I was then redirected to rebuild the codespace.

If you experience the same, hope this helps!

**Comment by Gh987654321** [comment]
Rubbish!  It's not that simplistic!  If it was, I wouldn't be here.  Already run - RESET GUI, RESET TERMINAL, UPDATED CODESPACE (7 TIMES!), REBUILD CODESPACE 3 TIMES, FULL REBUILD OF CODESPACE TWICE, FORCED SYNC CHANGES AND CLEANED UP REPOSITORY.  NOTHING WORKS.  THE LACK OF SUPPORT IS REALLY FRUSTRATING

**Comment by kallal79** [comment]
Codespace creation failed
Codespace creation failed. Codespace could not be created: Usage not allowed: 7908837174 cannot be billed for new codespaces...
